### PR TITLE
Remove glr parser

### PR DESCRIPTION
**Pull request by czpmango**
The previous implementation of pattern expression introduce bison glr parser to handle conflicts, but that is actually unnecessary and harmful. This pr refactors the implementation and fix some related bugs.

## What type of PR is this?
- [x] bug
- [ ] feature
- [x] enhancement

#### Issue(s) number: 
https://github.com/vesoft-inc/nebula/issues/5235

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [x] Documentation affected (Please add the label if documentation needs to be modified.)
- [x] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

### CODE REVIEW

File: tests/tck/features/match/Unwind.feature [hunk 1]
```diff
@@ -149,7 +149,7 @@ Feature: Unwind clause
     Then a SyntaxError should be raised at runtime: syntax error near `WITH'
     When executing query:
       """
-      MATCH (a:player {name:"Tim Duncan"}) - [e:like] -> (b)
+      MATCH (a:player {name:"Tim Duncan"})-[e:like]->(b)
```
Comment (Shylock-Hg) [inline tests/tck/features/match/Unwind.feature:152]: These are obviously unaccepted.

Comment (czpmango) [reply tests/tck/features/match/Unwind.feature:152]: I don't think it's a big deal.

Comment (czpmango) [reply tests/tck/features/match/Unwind.feature:152]: You also could write as this: MATCH (a:player {name:"Tim Duncan"}) -[e:like]-> (b)

File: src/parser/parser.yy [hunk 2]
```diff
@@ -1797,9 +1797,15 @@ match_path_list
...
-    : L_PAREN match_alias R_PAREN {
-        $$ = new MatchNode(*$2, nullptr, nullptr);
-        delete $2;
+    : L_PAREN R_PAREN {
+        $$ = new MatchNode();
+    }
+    | parenthesized_expression {
+        auto& e = $1;
+        if (e->kind() != Expression::Kind::kLabel) {
+            throw nebula::GraphParser::syntax_error(@1, "Invalid node pattern");
```
Comment (yixinglu) [inline src/parser/parser.yy:1806]: delete $1 before throw exception

**Review by yixinglu — COMMENTED** [review]
LGTM, prefer to test the case in #5221 and following tests:

```
with [1,2] as a, 2 as b, [1] as c return (a)-[b]-(c);
with [1,2] as a, 2 as b, [1] as c return (a)- [b] -(c);
```

**Review by yixinglu — APPROVED** [review]
Good job!

**Review by jievince — APPROVED** [review]

**Review by codesigner — APPROVED** [review]

**Comment by codesigner** [comment]
This fixed some patten expressibility broblems, but also introduce some compatibility(which is not very common), we should notice the change in release note when 3.5 released.

**Comment by Sophie-Xie** [comment]
> This fixed some patten expressibility problems, but also introduce some compatibility(which is not very common), we should notice the change in release note when 3.5 released. @Sophie-Xie

@MuYiYong pls review this incompatible pr, thanks.

File: src/parser/parser.yy (hunk 2 above)
Comment (czpmango) [reply src/parser/parser.yy:1807]: The expression memory is managed by ObjectPool, and there is no need for delete, which would result in double free.

**Pull request merged**